kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/templates/kroxylicious/KroxyliciousDeploymentTemplates.java

```diff
@@ -55,10 +55,8 @@ public static DeploymentBuilder defaultKroxyDeployment(String namespaceName, Str
                 .withLabels(kroxyLabelSelector)
                 .endMetadata()
                 .withNewSpec()
-                .withContainers(new ContainerBuilder()
-                        .withName("kroxylicious")
-                        .withImage(containerImage)
-                        .withImagePullPolicy("Always")
+                .withContainers(ContainerTemplates.baseImageBuilder("kroxylicious", containerImage)
+                        .withImagePullPolicy(Constants.PULL_IMAGE_ALWAYS)
```

Comment:
Kroxy was explicitly set to Always, should we keep this or use the once-per-process `baseImageBuilder` logic?

Reply:
I'd like to say switch but it seems like extra complexity and foot guns for an illusionary(?) perf gain

Reply:
Yeah, should mention that the first system test run with pull policy Always on all the test-clients containers runtime was the same as the previous build at 27 minutes.

Reply:
Yeah, the code isn't about perf, it's more about a consistent way to get the container pulled at least once for local development.

Reply:
> edit: ah yeap I see, the once-per-process thing is an optimization, maybe it could just be 'Always' for any '*latest*' images.

I think thats a healthy compromise. Though why are we running tests against Latest at all? Given we've just built an image we should be pulling by sha to ensure we test the thing wot we just built

Reply:
Looks like in `strimzi-ci` we use a stable identifier like `-t quay.io/the/repo:6f5c162` where the tag is the short commit hash from the PR. So the `Always` pull might be irrelevant in CI.

When I run locally it's defaulting to `quay.io/kroxylicious/kroxylicious:0.7.0-SNAPSHOT` so I guess something built by the last main merge? You'd need to set some environment variables up to get it to test a build you've pushed onto quay.io yourself. The image pull policy `Always` would be more useful in this kind of inner loop if you're trying to push changes to quay and then quickly see how it affects a system test.

Reply:
Just to confirm that in `strinzi-ci` we have two pipelines:
- PR: run against the `tag` of the latest commit done into the PR
- Merge: run against current latest SNAPSHOT, that is the current `main` branch after merging the PR.

Reply:
Sounds good `Always` is the right choice for the inner loop